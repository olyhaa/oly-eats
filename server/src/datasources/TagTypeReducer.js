export const tagTypeReducer = (tagType) => {
  if (!tagType) {
    return null;
  }
  return {
    id: tagType.id,
    label: tagType.label,
  };
};

export const tagTypeMutationReducer = ({
  success = false,
  message = undefined,
  tagType = null,
}) => {
  return {
    success,
    message,
    tagType: tagTypeReducer(tagType),
  };
};
