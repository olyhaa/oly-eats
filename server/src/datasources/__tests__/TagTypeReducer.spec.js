import { tagTypeReducer, tagTypeMutationReducer } from '../TagTypeReducer';

const mockTagTypes = [
  {
    id: '101',
    label: 'category',
  },
  {
    id: '102',
    label: 'cuisine',
  },
  {
    id: '103',
    label: 'meal_type',
  },
];

describe('tag type reducer', () => {
  it('properly transforms tag type object', () => {
    expect(tagTypeReducer(mockTagTypes[0])).toMatchSnapshot();
  });
});

describe('tag type mutation reducer', () => {
  it('properly transforms tag type object', () => {
    expect(
      tagTypeMutationReducer({ tagType: mockTagTypes[0] })
    ).toMatchSnapshot();
  });

  it('properly transforms tag type object on success', () => {
    expect(
      tagTypeMutationReducer({
        success: true,
        tagType: mockTagTypes[0],
      })
    ).toMatchSnapshot();
  });

  it('properly transforms tag type object on success with message', () => {
    expect(
      tagTypeMutationReducer({
        success: true,
        message: 'great job',
        tagType: mockTagTypes[0],
      })
    ).toMatchSnapshot();
  });

  it('properly transforms on success with message', () => {
    expect(
      tagTypeMutationReducer({ success: true, message: 'great job' })
    ).toMatchSnapshot();
  });

  it('properly transforms on failure', () => {
    expect(tagTypeMutationReducer({ success: false })).toMatchSnapshot();
  });

  it('properly transforms on failure with message', () => {
    expect(
      tagTypeMutationReducer({ success: false, message: 'bad test' })
    ).toMatchSnapshot();
  });
});
