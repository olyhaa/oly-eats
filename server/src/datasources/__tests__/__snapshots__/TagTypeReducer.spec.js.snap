// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tag type mutation reducer properly transforms on failure 1`] = `
Object {
  "message": undefined,
  "success": false,
  "tagType": null,
}
`;

exports[`tag type mutation reducer properly transforms on failure with message 1`] = `
Object {
  "message": "bad test",
  "success": false,
  "tagType": null,
}
`;

exports[`tag type mutation reducer properly transforms on success with message 1`] = `
Object {
  "message": "great job",
  "success": true,
  "tagType": null,
}
`;

exports[`tag type mutation reducer properly transforms tag type object 1`] = `
Object {
  "message": undefined,
  "success": false,
  "tagType": Object {
    "id": "101",
    "label": "category",
  },
}
`;

exports[`tag type mutation reducer properly transforms tag type object on success 1`] = `
Object {
  "message": undefined,
  "success": true,
  "tagType": Object {
    "id": "101",
    "label": "category",
  },
}
`;

exports[`tag type mutation reducer properly transforms tag type object on success with message 1`] = `
Object {
  "message": "great job",
  "success": true,
  "tagType": Object {
    "id": "101",
    "label": "category",
  },
}
`;

exports[`tag type reducer properly transforms tag type object 1`] = `
Object {
  "id": "101",
  "label": "category",
}
`;
